repo/object/object_reader.go

```diff
@@ -102,7 +106,38 @@ func (r *objectReader) Read(buffer []byte) (int, error) {
 	return readBytes, nil
 }
 
+func (r *objectReader) readAhead(ctx context.Context) {
+	if r.opts.ReadAheadBytes <= 0 {
+		return
+	}
+
+	var (
+		contentIDs  []content.ID
+		totalLength int64
+	)
+
+	for i := r.currentChunkIndex + 1; i < len(r.seekTable); i++ {
+		if cid, _, ok := r.seekTable[i].Object.ContentID(); ok {
+			if i > r.maxReadAheadChunkIndex {
+				contentIDs = append(contentIDs, cid)
+				r.maxReadAheadChunkIndex = i
+			}
+
+			totalLength += r.seekTable[i].Length
+			if totalLength > r.opts.ReadAheadBytes {
+				break
+			}
+		}
+	}
+
+	if len(contentIDs) > 0 {
+		go r.cr.PrefetchContents(ctx, contentIDs, "blobs")
```

Comment:
Each call to `readAhead()` will lead to creation of five goroutines for parallel fetch with their own buffers.  We should share the prefetch pool of workers and a cyclic buffer with each call to `PrefetchContents()`.

Reply:
Sharing buffers is already done using gather buffer, i will add a semaphore around goroutines.